iwyu_test_util.py

```diff
@@ -73,12 +74,56 @@ class Features:
   # Parses the output of 'include-what-you-use -print-targets'.
   SUPPORTED_TARGET_RE = re.compile(r'^\s+([a-z0-9_-]+)\s* - .*$')
 
+  # Parses the output of 'include-what-you-use -dM -E'.
+  PREPROCESSOR_MACRO_RE = re.compile(r'^#define ([A-z_][A-z0-9_]*)')
+
   def __init__(self):
     # Maps all known features names to prerequisite predicates.
     self.known = {
-      'has-target': Features.HasTarget
+      'has-target': Features.HasTarget,
+      'using-cstdlib': Features.UsingCstdlib,
+      'using-cxxstdlib': Features.UsingCXXstdlib,
```

Comment:
I'm still not convinced about the `using-` naming here. I would like to try and come up with some alternatives and compare and contrast. Just throwing some out here, I haven't thought about them much:

* `with-...stdlib`
* `active-...stdlib`
* `...stdlib`

More welcome :)

Reply:
Actually, it occurred to that maybe we should make it even more primitive: `has-predefined-macro()`?

But hmm, yeah, that doesn't work with the symbols in headers (provided by `<cstddef>` and `<stddef.h>` respectively). Worth considering something more basic at least, if there's some other way of achieving that.

Reply:
I'm happy to expose some preprocessor checks. Maybe something like
* `file-defines("<header>", "BLAH")` - 2 args though
* `has-predefined-macro("BLAH")`
Just thinking having a think how they read with IWYU_UNSUPPORTED and IWYU_REQUIRED...

    IWYU_REQUIRED: file-defines("abc.h", "FOO")
    IWYU_REQUIRED: has-predefined-macro("BAR")
    IWYU_UNSUPPORTED: file-defines("abd.c", "FOO")
    IWYU_UNSUPPORTED: has-predefined-macro("BAR")
    IWYU_XFAIL: file-defines("abc.h", "FOO")
    IWYU_XFAIL: has-predefined-macro("BAR")

I think we still want to encapsulate the standard library prvenance checks. `active-...stdlib` sounds reasonable.

Reply:
I'm on the fence about `active-`, I mostly suggested it to stimulate other ideas :).

I think `selected-` might look nice. Or to rhyme better with `has-target`, how about `uses-`?